src/index.js

```diff
@@ -24,6 +24,11 @@ const isNestedArray = arr => bool(
 
 const isMap = m => bool(Object.prototype.toString.call(m) === '[object Map]')
 
+// check if consumer is yielding our effect to immediatly cause the generator function to throw an error
+const shouldThrowError = obj => bool(obj && Object.keys(obj).includes('@THROW'))
+
+const throwError = message => ({ '@THROW': message })
```

Comment:
Since this `throwError` effect is only being used internally by this project, I would prefer to not add an extraneous properties to the object that aren't being used for anything. 

Is the value of matching the internal conversation of `redux-saga` to make it more clear that this is an imitation of a saga effect?